app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,21 @@
+class CommentsController < ApplicationController
+  def create
+    @post = Post.find(params[:post_id])
+    @user = User.find(params[:user_id])
+    @comment = @post.comments.new(comment_params)
+    @comment.author = current_user
+
+    if @comment.save
+      redirect_to user_post_url(@user, @post), notice: 'Comment created successfully!'
+    else
+      flash[:errors] = @comment.errors.full_messages
+      redirect_to user_post_url(@user, @post)
+    end
+  end
```

Comment:
- [ ] As required for the project, kindly ensure that A comment is created on behalf of the currently logged in user (current_user) of the app. The comments controller is to have have `new` and `create` actions like you did on in the post controller.below is an example:
```ruby
class CommentsController < ApplicationController
  ...

  def new
    @comment = Comment.new
  end

  def create
    @comment = @current_user.comments.new(params)
    @comment.post_id = params[:post_id]

    if @comment.save
      redirect_to post_path(@comment.post)
    else
      @article = Post.find(params[:post_id])
      redirect_to post_path(@post)
    end
  end
  ...
end
```